File: examples/sample/user_service.rb
require 'set'

module Sample
  class UserService
    def find_by_id(id)
      render(load_user(id))
    end

    def render_user(id)
      find_by_id(id)
    end

    private

    def load_user(id)
      "user##{id}"
    end

    def render(user)
      "[#{user}]"
    end
  end
end

File: examples/sample/UserService.php
<?php

namespace Sample;

use Sample\Support\Logger;

class UserService
{
    public function findById(int $id): string
    {
        return $this->render($this->loadUser($id));
    }

    public function renderUser(int $id): string
    {
        return $this->findById($id);
    }

    private function loadUser(int $id): string
    {
        return "user#" . $id;
    }

    private function render(string $user): string
    {
        return "[" . $user . "]";
    }
}
